test/BaseOracle.t.sol

```diff
@@ -0,0 +1,910 @@
+        oracle.startProcessing();
+        assertEq(oracle.getLastProcessingRefSlot(), refSlot1);
+    }
+
+    function test_submitConsensusReport_RevertsIfDeadlinePassed() public {
+        oracle.mockTime(deadline + 1);
+        vm.prank(address(consensus));
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                BaseOracle.ProcessingDeadlineMissed.selector,
+                deadline
+            )
+        );
+        oracle.submitConsensusReport(
+            keccak256("HASH_1"),
+            initialRefSlot,
+            deadline
+        );
+    }
+
+    function test_submitConsensusReport_RevertsIfNotCalledByConsensusContract()
+        public
+    {
+        vm.expectRevert(BaseOracle.SenderIsNotTheConsensusContract.selector);
+        oracle.submitConsensusReport(
+            keccak256("HASH_1"),
+            initialRefSlot,
+            deadline
+        );
+    }
+
+    function test_submitConsensusReport_RevertsIfZeroHash() public {
+        vm.prank(address(consensus));
+        vm.expectRevert(BaseOracle.HashCannotBeZero.selector);
+        oracle.submitConsensusReport(bytes32(0), initialRefSlot, deadline);
+    }
+
+    function test_submitConsensusReport_RevertsIfSubmittingOlderReport()
+        public
+    {
+        vm.prank(address(consensus));
+        oracle.submitConsensusReport(
+            keccak256("HASH_1"),
+            initialRefSlot,
+            deadline
+        );
+
+        uint256 badSlot = initialRefSlot - 1;
+        vm.prank(address(consensus));
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                BaseOracle.RefSlotCannotDecrease.selector,
+                badSlot,
+                initialRefSlot
+            )
+        );
+        oracle.submitConsensusReport(keccak256("HASH_1"), badSlot, deadline);
+    }
+
+    function test_submitConsensusReport_RevertsIfResubmitAlreadyProcessingReport()
+        public
+    {
+        vm.prank(address(consensus));
+        oracle.submitConsensusReport(
+            keccak256("HASH_1"),
+            initialRefSlot,
+            deadline
+        );
+
+        oracle.startProcessing();
+
+        vm.prank(address(consensus));
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                BaseOracle.RefSlotMustBeGreaterThanProcessingOne.selector,
+                initialRefSlot,
+                initialRefSlot
+            )
+        );
+        oracle.submitConsensusReport(
+            keccak256("HASH_1"),
+            initialRefSlot,
+            deadline
+        );
+    }
+
+    function test_submitConsensusReport_SubmitsInitialReport() public {
+        uint256 beforeCallCount = oracle.getConsensusReportLastCall().callCount;
+        assertEq(beforeCallCount, 0);
+
+        vm.prank(address(consensus));
+        oracle.submitConsensusReport(
+            keccak256("HASH_1"),
+            initialRefSlot,
+            deadline
+        );
```

Comment:
It would be nice to have event checked or the report value in the mock.

Reply:
yeah added tons of missed checks